java/src/main/native/src/row_conversion.cu

```diff
@@ -0,0 +1,365 @@
+                  } else {
+                    atomicAnd_block(valid_int, ~(1 << int_bit_offset));
+                  }
+                } else {
+                  //printf("ATOMIC BIT SET %p at %d (%d %d)\n", valid_int, int_bit_offset, col_index, row_index);
+                  // It is valid so just set the bit
+                  atomicOr_block(valid_int, 1 << int_bit_offset);
+                }
+            } // end column loop
+        } // end row copy
+        // wait for the row_group to be totally copied into shared memory
+        __syncthreads();
+
+        // Step 2: Copy the data back out
+        // We know row_size is always aligned with and a multipel of int64_t;
+        int64_t * long_shared = reinterpret_cast<int64_t *>(shared_data);
+        int64_t * long_output = reinterpret_cast<int64_t *>(output_data);
+
+        cudf::size_type shared_input_index = threadIdx.x + (threadIdx.y * blockDim.x);
+        cudf::size_type shared_input_stride = blockDim.x * blockDim.y;
+        cudf::size_type row_index_end = ((row_group_index + 1) * rows_per_group);
+        if (row_index_end > num_rows) {
+            row_index_end = num_rows;
+        }
+        cudf::size_type num_rows_in_group = row_index_end - (row_group_index * rows_per_group);
+        cudf::size_type shared_length = row_size * num_rows_in_group;
+
+        cudf::size_type shared_input_end = shared_length/sizeof(int64_t);
+
+        cudf::size_type start_output_index = (row_size * row_group_index * rows_per_group)/sizeof(int64_t);
+
+        for (cudf::size_type shared_index = shared_input_index; 
+                shared_index < shared_input_end;
+                shared_index += shared_input_stride) {
+            long_output[start_output_index + shared_index] = long_shared[shared_index];
+        }
+        __syncthreads();
+        // Go for the next round
+    }
+}
+
+static bool is_fixed_width(const cudf::column_view & v) {
+    return cudf::is_fixed_width(v.type());
+}
+
+static inline int32_t align_offset(int32_t offset, std::size_t alignment) {
+    return (offset + alignment - 1) & ~(alignment - 1);
+}
+
+static std::unique_ptr<cudf::column> fixed_width_convert_to_rows(
+        const cudf::size_type start_row,
+        const cudf::size_type num_rows,
+        const cudf::size_type num_columns,
+        const cudf::size_type size_per_row,
+        std::unique_ptr<rmm::device_uvector<cudf::size_type>> & column_start,
+        std::unique_ptr<rmm::device_uvector<cudf::size_type>> & column_size,
+        std::unique_ptr<rmm::device_uvector<const int8_t *>> & input_data,
+        std::unique_ptr<rmm::device_uvector<const cudf::bitmask_type *>> & input_nm,
+        const cudf::scalar & zero,
+        const cudf::scalar & scalar_size_per_row,
+        cudaStream_t stream,
+        rmm::mr::device_memory_resource* mr
+        ) {
+    int64_t total_allocation = size_per_row * num_rows;
+    if (total_allocation > std::numeric_limits<int>::max()) {
```

Comment:
why not make this `CUDF_EXPECTS(total_allocation < std::numeric_limits<int>::max(), "Table is too large to fit!");`